### ISSUE TITLE
New entities have the same parent of recycled entities

### ISSUE THREAD

**Issue by popcron** [friflo/Friflo.Engine.ECS#29]
when a new created entity uses a recycled slot/position from a previously destroyed entity, it still has the same parent that the recycled entity did. i dont understand if entities are meant to be removed from their parents before deletion, or if the library should be able to handle this common case.

also, great library thank you!

this is the test code:
```cs
[Fact]
public void CreateThenDestroyEntityWithParent()
{
    using World world = new();
    Entity a = world.CreateEntity();
    Entity b = world.CreateEntity();
    a.AddChild(b);
    int id = b.Id;
    Entity parent = b.Parent;
    //a.RemoveChild(b); //<- this fixes it
    b.DeleteEntity();
    Entity c = world.CreateEntity();
    Assert.Equal(id, c.Id);
    Assert.NotEqual(c.Parent, parent);
}
```

**Comment by friflo** [comment]
@popcron 
thx for report and test case!

Fixed by: https://github.com/friflo/Friflo.Engine.ECS/commit/52fb5371406b50e60e46e96b8a72107c45e413dd

**Comment by friflo** [comment]
Fixed in 3.0.0-preview.14
